manifests/plugin/dns_linode.pp

```diff
@@ -0,0 +1,60 @@
+# @summary Installs and configures the dns-linode plugin
+#
+# This class installs and configures the Let's Encrypt dns-linode plugin.
+# https://certbot-dns-linode.readthedocs.io
+#
+# @param package_name The name of the package to install when $manage_package is true.
+# @param api_key
+#   Optional string, linode api key value for authentication.
+# @param version
+#   string, linode api version.
+# @param config_path The path to the configuration directory.
+# @param manage_package Manage the plugin package.
+# @param propagation_seconds Number of seconds to wait for the DNS server to propagate the DNS-01 challenge.
+#
+class letsencrypt::plugin::dns_linode (
+  Optional[String[1]] $package_name = undef,
+  Optional[String[1]] $api_key      = undef,
+  String[1] $version      = '4',
+  Stdlib::Absolutepath $config_path = "${letsencrypt::config_dir}/dns-linode.ini",
+  Boolean $manage_package           = true,
+  Integer $propagation_seconds      = 10,
+) {
+  include letsencrypt
+
+  if ! $api_key {
+    fail('No authentication method provided, please specify api_key and version.')
+  }
```

Comment:
I guess this was based on the cloudflare plugin. I wonder if it would be better to make the `$api_key` parameter required rather than have this check, like how the dns_rfc2136 does it.